jwql/instrument_monitors/common_monitors/readnoise_monitor.py

```diff
@@ -0,0 +1,625 @@
+            self.substrt2 = header['SUBSTRT2']
+            self.subsize1 = header['SUBSIZE1']
+            self.subsize2 = header['SUBSIZE2']
+            self.date_obs = header['DATE-OBS']
+            self.time_obs = header['TIME-OBS']
+            self.expstart = '{}T{}'.format(self.date_obs, self.time_obs)
+        except KeyError as e:
+            logging.error(e)
+
+    def identify_tables(self):
+        """Determine which database tables to use for a run of the readnoise
+        monitor.
+        """
+
+        mixed_case_name = JWST_INSTRUMENT_NAMES_MIXEDCASE[self.instrument]
+        self.query_table = eval('{}ReadnoiseQueryHistory'.format(mixed_case_name))
+        self.stats_table = eval('{}ReadnoiseStats'.format(mixed_case_name))
+
+    def image_to_png(self, image, outname):
+        """Outputs an image array into a png file.
+
+        Parameters
+        ----------
+        image : numpy.ndarray
+            2D image array.
+
+        outname : str
+            The name given to the output png file.
+
+        Returns
+        -------
+        output_filename : str
+            The full path to the output png file.
+        """
+
+        output_filename = os.path.join(self.data_dir, '{}.png'.format(outname))
+
+        # Get image scale limits
+        z = ZScaleInterval()
+        vmin, vmax = z.get_limits(image)
+
+        # Plot the image
+        plt.figure(figsize=(12,12))
+        im = plt.imshow(image, cmap='gray', origin='lower', vmin=vmin, vmax=vmax)
+        plt.colorbar(im, label='Readnoise Difference (most recent dark - reffile) [DN]')
+        plt.title('{}'.format(outname))
+
+        # Save the figure
+        plt.savefig(output_filename, bbox_inches='tight', dpi=200, overwrite=True)
+        set_permissions(output_filename)
+        logging.info('\t{} created'.format(output_filename))
+
+        return output_filename
+
+    def make_crds_parameter_dict(self):
+        """Construct a paramter dictionary to be used for querying CRDS
+        for the current reffiles in use by the JWST pipeline.
+
+        Returns
+        -------
+        parameters : dict
+            Dictionary of parameters, in the format expected by CRDS.
+        """
+
+        parameters = {}
+        parameters['INSTRUME'] = self.instrument.upper()
+        parameters['DETECTOR'] = self.detector.upper()
+        parameters['READPATT'] = self.read_pattern.upper()
+        parameters['SUBARRAY'] = self.subarray.upper()
+        parameters['DATE-OBS'] = datetime.date.today().isoformat()
+        current_date = datetime.datetime.now()
+        parameters['TIME-OBS'] = current_date.time().isoformat()
+
+        return parameters
+
+    def make_histogram(self, data):
+        """Creates a histogram of the input data and returns the bin centers 
+        and the counts in each bin.
+
+        Parameters
+        ----------
+        data : numpy.ndarray
+            The input data.
+
+        Returns
+        -------
+        n : numpy.ndarray
```

Comment:
I would suggest to rename this variable to something like `counts`, which is much more explicit than `n`.

Reply:
I changed the variable name in this function to counts - I wasn't sure if you were also wanting all the references to `n` changed as well, e.g. the `amp1_n` database column, but let me know if so and I'll make those changes as well.